lib/datadog/statsd/message_buffer.rb

```diff
@@ -20,9 +20,21 @@ def initialize(connection,
 
         @buffer = String.new
         @message_count = 0
+
+        # store the pid for which this message buffer has been created
+        update_fork_pid
       end
 
       def add(message)
+        # we are in a new PID, which means the parent process has just forked and
+        # we are currently running in the child: we have to clean the buffer since
+        # we don't want to process/flush the metrics buffered by the parent process.
+        if forked?
+          buffer.clear
+          @message_count = 0
```

Comment:
Minor: I suggest extracting these to a `reset` that could also be called from `#initialize`. That way it's harder for changes to the class to cause drift and for us to forget to update this condition.